collect_app/src/main/java/org/odk/collect/android/widgets/AudioController.java

```diff
@@ -0,0 +1,238 @@
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.media.MediaPlayer;
+import android.net.Uri;
+import android.os.Handler;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.ImageButton;
+import android.widget.SeekBar;
+import android.widget.TextView;
+
+import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.R;
+import org.odk.collect.android.activities.FormEntryActivity;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.utilities.DateTimeUtils;
+
+import java.io.File;
+import java.io.IOException;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import timber.log.Timber;
+
+import static android.view.View.GONE;
+
+public class AudioController implements SeekBar.OnSeekBarChangeListener {
+
+    private static final int SEEK_FORWARD_TIME = 5000; // 5 seconds
+    private static final int SEEK_BACKWARD_TIME = 5000; // 5 seconds
+
+    @BindView(R.id.currentDuration)
+    TextView currentDurationLabel;
+    @BindView(R.id.totalDuration)
+    TextView totalDurationLabel;
+    @BindView(R.id.playBtn)
+    ImageButton playButton;
+    @BindView(R.id.seekBar)
+    SeekBar seekBar;
+
+    private Context context;
+    private MediaPlayer mediaPlayer;
+    private Handler seekHandler = new Handler();
+
+    /**
+     * Background Runnable thread
+     */
+    private Runnable updateTimeTask = new Runnable() {
+        public void run() {
+            try {
+                if (mediaPlayer.isPlaying()) {
+                    updateTimer();
+                }
+                seekHandler.postDelayed(this, 100);
+            } catch (IllegalStateException e) {
+                seekHandler.removeCallbacks(updateTimeTask);
+                Timber.i(e, "Attempting to update timer when player is stopped");
+            }
+        }
+    };
+
+    private State state;
+    private FormEntryPrompt formEntryPrompt;
+    private View view;
+
+    AudioController() {
+    }
+
+    void init(Context context, MediaPlayer mediaPlayer, FormEntryPrompt formEntryPrompt) {
+        this.context = context;
+        this.mediaPlayer = mediaPlayer;
+        this.formEntryPrompt = formEntryPrompt;
+
+        initMediaPlayer();
+        initControlsLayout();
+    }
+
+    @OnClick(R.id.fastForwardBtn)
+    void fastForwardMedia() {
+        seekTo(mediaPlayer.getCurrentPosition() + SEEK_FORWARD_TIME);
+    }
+
+    @OnClick(R.id.fastRewindBtn)
+    void rewindMedia() {
+        seekTo(mediaPlayer.getCurrentPosition() - SEEK_BACKWARD_TIME);
+    }
+
+    /**
+     * Seeks media to the new position and updates timer
+     */
+    private void seekTo(int newPosition) {
+        if (newPosition < 0) {
+            newPosition = 0;
+        } else if (newPosition > mediaPlayer.getDuration()) {
+            newPosition = mediaPlayer.getDuration();
+        }
+
+        mediaPlayer.seekTo(newPosition);
```

Comment:
Seven lines can be one, with our friends min and max.

![screen shot 2018-03-18 at 8 39 42 pm](https://user-images.githubusercontent.com/55099/37577022-a7760ffe-2aec-11e8-9f3a-f52391571fa8.png)